### PR TITLE
Honor use_proxy parameter in uri module

### LINKED ISSUE #58632: use_proxy is documented but never used in uri module
##### SUMMARY
`use_proxy` is described as module parameter but is never used in the code. 

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
`uri` module

##### ANSIBLE VERSION
```
ansible 2.8.1
```

##### CONFIGURATION
```
-
```

##### OS / ENVIRONMENT
all

##### STEPS TO REPRODUCE

We use a corporate proxy, except for the desired host. So i put `use_proxy: no` into the task like this:
```
     - uri:
        url: "https://<example url>/"
        use_proxy: no
```

The result was the following error in output:
```
"msg": "Status code was -1 and not [200]: Request failed: <urlopen error Tunnel connection failed: 404 Not Found>"
```

I inspected the code:
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/net_tools/basics/uri.py

At line 160, the parameter is described. But this is the only place in this file. So i think, the parameter can not work.

##### EXPECTED RESULTS
`use_proxy: no` should work as intended like in `get_url`

##### ACTUAL RESULTS
`use_proxy: no` does not work.

### PR DESCRIPTION
**Pull request by Kassec**
##### SUMMARY
Fixes #58632

##### ISSUE TYPE
Bugfix Pull Request

##### COMPONENT NAME
uri

### CODE REVIEW

**Comment by samdoran** [comment]
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

**Review by jamescassell — APPROVED** [review]
shipit

File: changelogs/fragments/58632-uri-handle_use_proxy.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - uri - properly honor use_proxy parameter
```
Comment (samdoran) [inline changelogs/fragments/58632-uri-handle_use_proxy.yaml]:
```suggestion
  - uri - properly honor ``use_proxy`` parameter (https://github.com/ansible/ansible/issues/58632)
```

**Comment by bmillemathias** [comment]
Hi @Kassec 
Would you mind rebase your patch ?

**Comment by bcoca** [comment]
closing in favor of #77312

**Pull request closed without merging**